addons/mail/static/src/new/core/partner_model.js

```diff
@@ -4,6 +4,7 @@
  * @typedef Data
  * @property {number} id
  * @property {string} name
+ * @property {string} im_status
```

Comment:
im status is a fixed set of strings so you could document them like this: `@property {'online'|'offline'} im_status`